client/shaders/nodes_shader/opengl_vertex.glsl

```diff
@@ -75,12 +110,19 @@ float disp_z;
 		smoothTriangleWave(animationTimer * 13.0 + tOffset)) * 0.5;
 #endif
 
+	worldPosition = (mWorld * gl_Vertex).xyz;
 
 #if (MATERIAL_TYPE == TILE_MATERIAL_WAVING_LIQUID_TRANSPARENT || MATERIAL_TYPE == TILE_MATERIAL_WAVING_LIQUID_OPAQUE || MATERIAL_TYPE == TILE_MATERIAL_WAVING_LIQUID_BASIC) && ENABLE_WAVING_WATER
+	// Generate waves with Perlin-type noise.
+	// The constants are calibrated such that they roughly
+	// correspond to the old sine waves.
 	vec4 pos = gl_Vertex;
-	pos.y -= 2.0;
-	float posYbuf = (pos.z / WATER_WAVE_LENGTH + animationTimer * WATER_WAVE_SPEED * WATER_WAVE_LENGTH);
-	pos.y -= sin(posYbuf) * WATER_WAVE_HEIGHT + sin(posYbuf / 7.0) * WATER_WAVE_HEIGHT;
+	vec3 wavePos = worldPosition + cameraOffset;
+	wavePos.x /= WATER_WAVE_LENGTH * 3;
+	wavePos.z /= WATER_WAVE_LENGTH * 2;
```

Comment:
```suggestion
	wavePos.y /= WATER_WAVE_LENGTH * 2;
```

Reply:
Definitely not. The variation is on x and z (it's and 3d noise function)

Reply:
I just thought it was `y`, but I'm incorrect.
You can resolve the conversation; thank you.